lib/internal/per_context/primordials.js

```diff
@@ -12,17 +12,9 @@
 // `primordials.Object` where `primordials` is a lexical variable passed
 // by the native module compiler.
 
-const ReflectApply = Reflect.apply;
-
-// This function is borrowed from the function with the same name on V8 Extras'
-// `utils` object. V8 implements Reflect.apply very efficiently in conjunction
-// with the spread syntax, such that no additional special case is needed for
-// function calls w/o arguments.
-// Refs: https://github.com/v8/v8/blob/d6ead37d265d7215cf9c5f768f279e21bd170212/src/js/prologue.js#L152-L156
 function uncurryThis(func) {
-  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
+  return Function.prototype.call.bind(func);
 }
```

Comment:
To avoid accessing `Function.prototype.call.bind` from the global object every time `uncurryThis` is called, this should use the following:
```suggestion
const uncurryThis = Function.prototype.call.bind(Function.prototype.bind);
```

Reply:
Oh boy, the recursion is upon us 😅

Reply:
My head hurts reading this, but I think it might indeed be the most optimized. Thanks, let's try it.

Reply:
I've tried your original suggestion on my local machine, but it doesn't compile. Good catch regarding the use of `Function.prototype` which could be mutated by users indeed.

Reply:
I forgot it has to be:
```js
Function.prototype.call.bind(
	Function.prototype.bind,
	// I originally missed this second call:
	Function.prototype.call,
);
```